### PR TITLE
add api to get team name

### PR DESCRIPTION
**Pull request by ticoann**
add list of team names for assigning the request.
team names are only determined by agents registered to wmstats. (Not sure this will cause the problem).
Otherwise we have to have the interface to add the team name like in reqmgr1 
@vkuznet, @amaltaro, Please review

### CODE REVIEW

**Comment by ticoann** [comment]
REST api will be (not deployed)
https://cmsweb.cern.ch/wmstatsserver/data/teams
Test vm is set
https://reqmgr2-dev.cern.ch/wmstatsserver/data/teams

**Comment by vkuznet** [comment]
Seagnchan,
how this API will work? I mean does teams change over time or depends on
request? When should I query this API, every time when somebody
click on assign link, can we cache teams?

I think you really need to clarify the behavior for that.
So far it is just REST API which as far as I can tell will return
list of teams.

Thanks,
Valentin.

**Comment by ticoann** [comment]
Hi Valentin,
In production teams are not changed, we very occasionally add teams but in general we have only a few teams and doesn't change. Although call is pretty light weight and repeated call won't have much overhead but I think caching would be fine. Are you trying to caching in browser? I think some of the browser caches ajax call automatically,  you can do manual caching if you want.

FYI. team is something that related to agent.  Basically team is used to pull down the request to certain agent.  So if a team is assigned to given request only agent belongs that team can pull that request.
We were planning to remove team concept completely, since request should run all the agent due to the not perfect priority and resource control we maintain a couple of teams for different agent.

When agent gets deployed, it registers (actually just reports) the team in wmstats so that is why I created wmstats REST api.

**Comment by vkuznet** [comment]
Seangchan,
I changed https://github.com/dmwm/WMCore/pull/6681/
with respect to discussion we had. getTeams local API now fetch teams
all the time and fall backs into using cached values.
Please review, since I didn't test it on vm.
Best,
Valentin.

File: src/python/WMCore/Services/WMStats/WMStatsReader.py [hunk 1]
```diff
@@ -248,19 +248,19 @@ def _getAgentInfo(self):
         
         return result
     
-    def agentsByTeam(self, ignoreDrain = True):
+    def agentsByTeam(self, filterDrain=True):
```
Comment (amaltaro) [inline src/python/WMCore/Services/WMStats/WMStatsReader.py]: I'd default it to False instead.

File: src/python/WMCore/Services/WMStats/WMStatsReader.py [hunk 2]
```diff
@@ -248,19 +248,19 @@ def _getAgentInfo(self):
...
         """
         return a dictionary like {team:#agents,...}
         """
         result = self._getAgentInfo()
         response = dict()
+        import pprint
```
Comment (amaltaro) [inline src/python/WMCore/Services/WMStats/WMStatsReader.py]: you forgot this pprint here :)

File: src/python/WMCore/Services/WMStats/WMStatsReader.py (hunk 1 above)
Comment (ticoann) [inline src/python/WMCore/Services/WMStats/WMStatsReader.py]: Actually, in initial usage, we filtered out drain agent for calculating the resource distribution in AgentWatcher component. I can change those if it makes more sense including drain agent for team.

File: src/python/WMCore/Services/WMStats/WMStatsReader.py (hunk 2 above)
Comment (ticoann) [inline src/python/WMCore/Services/WMStats/WMStatsReader.py]: Thanks, updated

**Pull request merged**